app/delivery/send_to_providers.py

```diff
@@ -128,8 +135,32 @@ def check_file_url(file_info: Dict[str, str], notification_id: UUID):
         raise InvalidUrlException
 
 
+def check_for_malware_errors(document_download_response_code, notification):
+    """
+    Check verdict and download calls to the document-download-api will
+    return error codes if the scan is in progress or if malware was detected.
+    This function contains the logic for handling these errors.
+    """
+    # 423 "Locked" response is sent if malicious content was detected
+    if document_download_response_code == 423:
+        current_app.logger.info(
+            f"Malicious content detected! Download and attachment failed for notification.id: {notification.id}"
+        )
+        # Update notification that it contains malware
+        malware_failure(notification=notification)
+
+    # 428 "Precondition Required" response is sent if the scan is still in progress
+    if document_download_response_code == 428:
+        current_app.logger.info(f"Malware scan in progress, could not download files for notification.id: {notification.id}")
+        # Throw error so celery will retry in sixty seconds
+        raise MalwareScanInProgressException
+
+    if document_download_response_code != 200:
+        document_download_internal_error(notification=notification)
+
+
 def send_email_to_provider(notification: Notification):
-    current_app.logger.info(f"Sending email to provider for notification id {notification.id}")
+    current_app.logger.info(f"Sending email to provider for notification id: {notification.id}")
```

Comment:
A side note that we should probably check our existing queries in CloudWatch and TF when we change logs. Probably fine here I hope.

Reply:
Good point - I did not need to make that log change! I'll revert.